src/lando/api/legacy/workers/base.py

```diff
@@ -308,6 +313,59 @@ def update_repo(
             )
             raise PermanentFailureException(message) from e
 
+    def apply_patch(
+        self,
+        repo: Repo,
+        job: BaseJob,
+        scm: AbstractSCM,
+        revision: Revision,
+    ) -> None:
+        """Apply patches to repo with job status handling."""
+        try:
+            scm.apply_patch(
+                revision.diff,
+                revision.commit_message,
+                revision.author,
+                revision.timestamp,
+            )
+        except NoDiffStartLine as exc:
+            message = (
+                "Lando encountered a malformed patch, please try again. "
+                "If this error persists please file a bug: "
+                "Patch without a diff start line."
+            )
+            logger.error(message)
+            job.transition_status(
+                JobAction.FAIL,
+                message=message,
+            )
+            raise PermanentFailureException(message) from exc
+
+        except PatchConflict as exc:
+            breakdown = scm.process_merge_conflict(
+                repo.normalized_url, revision.revision_id, str(exc)
+            )
+            job.error_breakdown = breakdown
+
+            message = (
+                f"Problem while applying patch in revision {revision.revision_id}:\n\n"
+                f"{str(exc)}"
+            )
+            logger.exception(message)
+            job.transition_status(JobAction.FAIL, message=message)
+            raise PermanentFailureException(message) from exc
```

Comment:
Perhaps we could just except the parent Exception, `PatchApplicationFailure`, and maybe include the `exc.__doc__` (a trick I recently learned!) to extract the docstring of the concrete class and add it to the message?